core/src/main/java/com/linecorp/armeria/internal/server/annotation/KotlinUtil.java

```diff
@@ -0,0 +1,172 @@
+                    coroutineUtilClass, "callKotlinSuspendingMethod",
+                    MethodType.methodType(
+                            CompletableFuture.class,
+                            ImmutableList.of(Method.class, Object.class,
+                                             Object[].class, ExecutorService.class,
+                                             RequestContext.class))
+            );
+        } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException e) {
+            // ignore
+        } finally {
+            CALL_KOTLIN_SUSPENDING_METHOD = callKotlinSuspendingMethod;
+        }
+
+        Method isContinuation = null;
+        Method isSuspendingFunction = null;
+        Method isReturnTypeUnit = null;
+        try {
+            final Class<?> kotlinUtilClass =
+                    getClass("com.linecorp.armeria.internal.common.kotlin.KotlinUtil");
+
+            isContinuation = kotlinUtilClass.getMethod("isContinuation", Class.class);
+            isSuspendingFunction = kotlinUtilClass.getMethod("isSuspendingFunction", Method.class);
+            isReturnTypeUnit = kotlinUtilClass.getMethod("isReturnTypeUnit", Method.class);
+        } catch (ClassNotFoundException | NoSuchMethodException e) {
+            // ignore
+        } finally {
+            IS_CONTINUATION = isContinuation;
+            IS_SUSPENDING_FUNCTION = isSuspendingFunction;
+            IS_RETURN_TYPE_UNIT = isReturnTypeUnit;
+        }
+
+        boolean isKotlinReflectionPresent = false;
+        try {
+            getClass("kotlin.reflect.full.KClasses");
+            isKotlinReflectionPresent = true;
+        } catch (ClassNotFoundException e) {
+            // ignore
+        } finally {
+            IS_KOTLIN_REFLECTION_PRESENT = isKotlinReflectionPresent;
+        }
+
+        Class<? extends Annotation> metadataClass = null;
+        try {
+            metadataClass = (Class<? extends Annotation>) getClass("kotlin.Metadata");
+        } catch (ClassNotFoundException e) {
+            // ignore
+        } finally {
+            METADATA_CLASS = metadataClass;
+        }
+    }
+
+    /**
+     * Returns a method which invokes Kotlin suspending functions.
+     */
+    @Nullable
+    static MethodHandle getCallKotlinSuspendingMethod() {
+        return CALL_KOTLIN_SUSPENDING_METHOD;
+    }
+
+    /**
+     * Returns true if a method is written in Kotlin.
+     */
+    static boolean isKotlinMethod(Method method) {
+        return METADATA_CLASS != null &&
+               method.getDeclaringClass().getAnnotation(METADATA_CLASS) != null;
+    }
+
+    /**
+     * Returns true if a method is a suspending function.
+     */
+    static boolean isSuspendingFunction(Method method) {
+        try {
+            return isKotlinMethod(method) &&
+                   IS_KOTLIN_REFLECTION_PRESENT &&
+                   IS_SUSPENDING_FUNCTION != null &&
+                   (Boolean) IS_SUSPENDING_FUNCTION.invoke(null, method);
```

Comment:
Is it possible to return `boolean`? The returned value of `IS_CONTINUATION` is casted to `boolean`.